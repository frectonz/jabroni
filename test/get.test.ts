import { makeWebSocketFetch, nanoid, snapshotTest } from "./base.ts";

const $fetch = await makeWebSocketFetch({
  url: "ws://127.0.0.1:3030",
  connectionCount: 10,
});

Deno.test(
  "get single row",
  snapshotTest($fetch, {
    type: "GetRow",
    table: "employees",
    key: 1,
    select: [],
    request_id: nanoid(),
  }),
);

Deno.test(
  "get single row with selected columns",
  snapshotTest($fetch, {
    type: "GetRow",
    table: "employees",
    key: 1,
    select: ["Title", "FirstName", "LastName"],
    request_id: nanoid(),
  }),
);

Deno.test(
  "get single row from a table that does not exist",
  snapshotTest($fetch, {
    type: "GetRow",
    table: "cats",
    key: 1,
    select: ["Title", "FirstName", "LastName"],
    request_id: nanoid(),
  }),
);

Deno.test(
  "get single row from a with a non existent key",
  snapshotTest($fetch, {
    type: "GetRow",
    table: "employees",
    key: 10,
    select: ["Title", "FirstName", "LastName"],
    request_id: nanoid(),
  }),
);
