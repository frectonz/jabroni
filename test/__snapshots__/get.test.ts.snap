export const snapshot = {};

snapshot[`get single row 1`] = `
{
  data: {
    request_id: "",
    row: {
      Address: "11120 Jasper Ave NW",
      BirthDate: "1962-02-18 00:00:00",
      City: "Edmonton",
      Country: "Canada",
      Email: "andrew@chinookcorp.com",
      EmployeeId: 1,
      Fax: "+1 (780) 428-3457",
      FirstName: "Andrew",
      HireDate: "2002-08-14 00:00:00",
      LastName: "Adams",
      Phone: "+1 (780) 428-9482",
      PostalCode: "T5K 2N1",
      ReportsTo: null,
      State: "AB",
      Title: "General Manager",
    },
    table: "employees",
    type: "GetRow",
  },
}
`;

snapshot[`get single row with selected columns 1`] = `
{
  data: {
    request_id: "",
    row: {
      FirstName: "Andrew",
      LastName: "Adams",
      Title: "General Manager",
    },
    table: "employees",
    type: "GetRow",
  },
}
`;

snapshot[`get single row from a table that does not exist 1`] = `
{
  error: {
    table: "cats",
    type: "TableNotFound",
  },
}
`;

snapshot[`get single row from a with a non existent key 1`] = `
{
  error: {
    type: "RowNotFound",
  },
}
`;
